src/theory/quantifiers/single_inv_partition.h

```diff
@@ -0,0 +1,269 @@
+   *
+   * This initializes this class to check whether formula n is single
+   * invocation with respect to the input functions in funcs only.
+   * Below, the "processed formula" is a formula generated by this
+   * call that is equivalent to n (if this call is successful).
+   *
+   * This method returns true if all input functions have identical
+   * argument types, and false otherwise. Notice that all
+   * access functions below are only valid if this class is
+   * successfully initialized.
+   */
+  bool init(std::vector<Node>& funcs, Node n);
+  /** initialize this partition for formula n
+   *
+   * In contrast to the above method, this version assumes that
+   * all uninterpreted functions are input functions. That is, this
+   * method is equivalent to the above function with funcs containing
+   * all uninterpreted functions occurring in n.
+   */
+  bool init(Node n);
+  /** is the processed formula purely single invocation?
+   *
+   * A formula is purely single invocation if it is equivalent to:
+   *   t[ f1( x ), ..., fn( x ), x ],
+   * for some F, where f1...fn are the input functions.
+   * Notice that the free variables of t are exactly x.
+   */
+  bool isPurelySingleInvocation() { return d_conjuncts[1].empty(); }
+  /** is the processed formula non-ground single invocation?
+   *
+   * A formula is non-ground single invocation if it is equivalent to:
+   *   F[ f1( x ), ..., fn( x ), x, y ],
+   * for some F, where f1...fn are the input functions.
+   */
+  bool isNonGroundSingleInvocation()
+  {
+    return d_conjuncts[3].size() == d_conjuncts[1].size();
+  }
+  /** Get the (portion of) the processed formula that is single invocation
+   *
+   * Notice this method returns the anti-skolemized version of the input
+   * formula. In (EX1), this method returns:
+   *   z = g( y ) V z = b
+   * where z is the first-order variable for f (see
+   * getFirstOrderVariableForFunction).
+   */
+  Node getSingleInvocation() { return getConjunct(0); }
+  /** Get the (portion of) the processed formula that is not single invocation
+   *
+   * This formula and the above form a partition of the conjuncts of the
+   * processed formula, that is:
+   *   getSingleInvocation() * sigma ^ getNonSingleInvocation()
+   * is equivalent to the processed formula, where sigma is a substitution of
+   * the form:
+   *   z_1 -> f_1( x ) .... z_n -> f_n( x )
+   * where z_i are the first-order variables for input functions f_i
+   * for all i=1,...,n, and x are the single invocation arguments of the input
+   * formulas (see d_si_vars).
+   */
+  Node getNonSingleInvocation() { return getConjunct(1); }
+  /** get full specification
+   *
+   * This returns getSingleInvocation() * sigma ^ getNonSingleInvocation(),
+   * which is equivalent to the processed formula, where sigma is the
+   * substitution described above.
+   */
+  Node getFullSpecification() { return getConjunct(2); }
+  /** get first order variable for input function f
```

Comment:
Make docstrings consistent. The above have a newline between the brief and the description.